src/lsp/MessageIO.ts

```diff
@@ -129,6 +143,17 @@ export class MessageIOReader extends AbstractMessageReader implements MessageRea
 			this.nextMessageLength = -1;
```

Comment:
I don't think this is a good idea to inject code here to process custom messages here.

Why not do that in `MessageIO.on_message_callback` ?